imports/methods/TypedMethod.ts

```diff
@@ -47,29 +83,29 @@ export default class TypedMethod<
         but the actual Meteor audit-argument-checks package will still fire if
         we don't use check methods. */
       [this.name](arg0: Args) {
-        self.execute(this, arg0);
+        return self.execute(this, arg0);
       },
     });
   }
 
-  execute(context: Meteor.MethodThisType, arg0: Args): Return {
+  execute(context: Meteor.MethodThisType, arg0: Args): Return | Promise<Return> {
     if (!this.definition) {
       throw new Error(`TypedMethod ${this.name} has not been defined`);
     }
 
+    // In the case of no arguments, the type system will track the return value
+    // from `validate` as `void`, but because it's a function that doesn't
+    // return anything, it will in practice be `undefined`.
     const validatedArgs = this.definition.validate.bind(context)(arg0);
-    const result = this.definition.run.bind(context)(validatedArgs);
+    const result = this.definition.run.bind(context)(validatedArgs as any);
     return result;
   }
 
-  call(
-    arg0: Args,
-    callback?: (error: Meteor.Error, result: Return) => void,
-  ): void {
-    return Meteor.call(this.name, arg0, callback);
+  call(...args: TypedMethodCallArgs<Args, Return>): void {
```

Comment:
I'll add a comment. FWIW, I view this as a good thing, since I think we should start leaning towards preferring promises over blocking fibers on the server anyway, and `callPromise` is right there if you need it.

Reply:
Err actually that's not true - the callback is an optional argument (for both void and non-void) in `TypedMethodCallArgs`, and if not present, it'll trigger the default Meteor behavior.

Reply:
I think my claim here was that the `void` return type from the `call` function is not strictly correct in the fiber-blocking case, but that we don't really care about that

Reply:
Ah that is true. It's...fixable but yeah not worth it.